Ensure that the email input has a unique Id

B77Mills opened the pull request:
This add support of passing in a id to email.vue component in order to ensure that the login email input is unique.  By default when calling the email.vue component on its own it will still use sign-on-email as the id.
<img width="487" alt="Screen Shot 2022-06-09 at 9 43 02 AM" src="https://user-images.githubusercontent.com/3845869/172875341-c0d6893d-287f-4f5b-97a6-4f36c5384bb4.png">

B77Mills committed:
Ensure that the email input has a unique Id

This add support of passing in a id to email.vue component in order to ensure that the login email input is unique.  By default when calling the email.vue component on its own it will still use sign-on-email as the id.

Shinsina commented on packages/marko-web-identity-x/browser/form/fields/email.vue:
```diff
@@ -42,10 +42,11 @@ export default {
     },
+    id: {
+      type: String,
+      default: 'sign-on-email'
```
Missing trailing comma.

Shinsina commented on packages/marko-web-identity-x/browser/login.vue:
```diff
@@ -167,6 +169,7 @@ export default {
    *
    */
   mounted() {
+    this.loginInputId = `${this.loginInputId}-${this._uid}`;
```
The linter doesn't seem to like _uid (causing the Travis build to fail), either add the appropriate eslint-ignore/disable comment to the line prior to this or try using dynamic property access ```this['_uid']```

B77Mills committed:
Add uniqueId generator and use in email.vue

Add uniqueId generator util and use it in email.vue to ensure the email label/input have unique Id.  Used https://github.com/lukeed/uid as a reference to the uid generator function.

Shinsina commented on packages/marko-web-identity-x/browser/utils/uid.js:
```diff
@@ -0,0 +1,20 @@
+  let i = 0;
+  const tmp = (len || 11);
+  if (!buffer || ((idx + tmp) > size * 2)) {
+    for (buffer = '', idx = 0; i < size; i++) {
```
Should be able to do i += 1 here and remove the eslint-disable no-plusplus

zarathustra323 commented on packages/marko-web-identity-x/browser/form/fields/email.vue:
```diff
@@ -44,7 +45,7 @@ export default {
+    id: '',
   }),
   computed: {
     email: {
```
```js
id() {
  return elementId('identity-x-email-field');
},
```

zarathustra323 commented on packages/marko-web-identity-x/browser/form/fields/email.vue:
```diff
@@ -19,6 +19,7 @@
 <script>
 import FormGroup from '../common/form-group.vue';
```
```js
import elementId from '@parameter1/base-cms-marko-web/browser/components/element-id';
```

B77Mills committed:
Use `base-cms-marko-web/browser/components/element-id`

Use base-cms-marko-web/browser/components/element-id instead of a new uid function
